www/js/app.js

```diff
@@ -1290,17 +1353,29 @@ function handleIframeDrop (e) {
     e.preventDefault();
 }
 
-function handleFileDrop (packet) {
+async function handleFileDrop (packet) {
     packet.stopPropagation();
     packet.preventDefault();
     configDropZone.style.border = '';
     var files = packet.dataTransfer.files;
-    document.getElementById('openLocalFiles').style.display = 'none';
+    document.getElementById('selectInstructions').style.display = 'none';
+    document.getElementById('fileSelectionButtonContainer').style.display = 'none';
     document.getElementById('downloadInstruction').style.display = 'none';
     document.getElementById('selectorsDisplay').style.display = 'inline';
-    setLocalArchiveFromFileList(files);
-    // This clears the display of any previously picked archive in the file selector
     document.getElementById('archiveFiles').value = null;
+
+    // value will be set to true if a folder is dropped then there will be no need to
+    // call the `setLocalArchiveFromFileList`
+    let loadZim = true;
+
+    if (params.isFileSystemApiSupported) loadZim = await abstractFilesystemAccess.handleFolderDropViaFileSystemAPI(packet)
```

Comment:
Is file drop supported by Firefox I don't think there is any file drop. If Filedrop is supported then I need to add the same conditions here as well

Reply:
Do you mean drag-and-drop? Firefox supports this, so long as you haven't turned it off in Expert Settings! However, I'm not sure if it supports dragging a folder in. It supports dragging one or more files in the same way as picking one or more files with the (legacy) File picker. Files dragged act in exactly the same way as if they had been picked.

It's possible that a folder dragged in has the webkitdirectory property, in which case you could use it just like picking a folder in Firefox. I didn't implement that in the PWA, because it seemed unlikely to me that anyone would drag a folder as opposed to dragging some files from the folder.

Reply:
File dragging is indeed broken in Firefox in your implementation, so that will need to be fixed, as it works in main.

Just treat the file as a File object and use the existing function for that (can't' remember if it's selectArchiveFromFileList or something). Or check how it's done on main.

Dragging a folder works fine in Firefox in your implementation.